app/domain/operations/hbx_enrollments/drop_enrollment_members.rb

```diff
@@ -11,51 +11,101 @@ class DropEnrollmentMembers
+        return Failure('Not an ivl enrollment.') unless params[:hbx_enrollment].is_ivl_by_kind? # Added per the request, also need further development if to be used for shop
         return Failure('No members selected to drop.') if params[:options].select{|string| string.include?("terminate_member")}.empty?
         return Failure('No termination date given.') if params[:options].select{|string| string.include?("termination_date")}.empty?
+        return Failure('Termination date cannot be outside of the current calender year.') unless Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y").year == TimeKeeper.date_of_record.year
 
         Success(params)
       end
 
       def drop_enrollment_members(params)
-        terminate_date = Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y")
+        termination_date = Date.strptime(params[:options]["termination_date_#{params[:hbx_enrollment].id}"], "%m/%d/%Y")
         dropped_enr_members = params[:options].select{|string| string.include?("terminate_member")}.values
         all_enr_members = params[:hbx_enrollment].hbx_enrollment_members
-        eligible_members = all_enr_members.reject{ |member| dropped_enr_members.include?(member.id.to_s) }
-
-        reinstatement = Enrollments::Replicator::Reinstatement.new(params[:hbx_enrollment], terminate_date + 1.day, 0, eligible_members).build
-        reinstatement.save!
+        non_eligible_members = all_enr_members.select{ |member| dropped_enr_members.include?(member.id.to_s) }
+        new_effective_date = (termination_date > params[:hbx_enrollment].effective_on) ? termination_date + 1.day : params[:hbx_enrollment].effective_on
+        return Failure('Select termination date that would result member drop in present calender year.') unless new_effective_date.year == termination_date.year
 
-        get_household_applied_aptc(reinstatement, terminate_date + 1.day) if params[:hbx_enrollment].applied_aptc_amount > 0
+        custom_params = custom_params(params[:hbx_enrollment], new_effective_date)
+        clone_enrollment = Operations::HbxEnrollments::Clone.new.call({hbx_enrollment: params[:hbx_enrollment], effective_on: new_effective_date, options: custom_params})
+        return clone_enrollment if clone_enrollment.failure?
 
-        reinstatement.force_select_coverage! if reinstatement.may_reinstate_coverage?
-        reinstatement.begin_coverage! if reinstatement.may_begin_coverage? && reinstatement.effective_on <= TimeKeeper.date_of_record
+        new_enrollment = clone_enrollment.value!
+        new_enrollment.hbx_enrollment_members.delete_if {|mem| non_eligible_members.pluck(:applicant_id).include?(mem.applicant_id)}
+        set_product_id(params[:hbx_enrollment], new_enrollment)
+        new_enrollment.check_for_subscriber
```

Comment:
We also need to reset consumer_role_id(to new subscriber) on dep drop enrollment, when subscriber is dropped from base enrollment.